Find a functionally equivalent code:android.app.ApplicationErrorReport.CrashInfo.writeToParcel:COMMENT
Method Modifier: public      
Comment:/**
 * Save a CrashInfo instance to a parcel.
 */

@@ -1,4 +1,5 @@
 {
+    int start = dest.dataPosition();
     dest.writeString(exceptionClassName);
     dest.writeString(exceptionMessage);
     dest.writeString(throwFileName);
@@ -6,9 +7,20 @@
     dest.writeString(throwMethodName);
     dest.writeInt(throwLineNumber);
     dest.writeString(stackTrace);
+    int total = dest.dataPosition() - start;
+    if (total > 20 * 1024) {
+        Slog.d("Error", "ERR: exClass=" + exceptionClassName);
+        Slog.d("Error", "ERR: exMsg=" + exceptionMessage);
+        Slog.d("Error", "ERR: file=" + throwFileName);
+        Slog.d("Error", "ERR: class=" + throwClassName);
+        Slog.d("Error", "ERR: method=" + throwMethodName + " line=" + throwLineNumber);
+        Slog.d("Error", "ERR: stack=" + stackTrace);
+        Slog.d("Error", "ERR: TOTAL BYTES WRITTEN: " + (dest.dataPosition() - start));
+    }
 }

